libc/src/string/memmove.c

```diff
@@ -0,0 +1,5 @@
+#include <string.h>
+
+void *memmove(void *dest, const void *src, size_t n){
```

Comment:
memmove is not the same as memcpy. memcpy has the requirement that the src and dst do not overlap, memmove does not have this requirement and is safe to use on overlapping ranges.

Reply:
Fixed using another buffer array

Reply:
> Fixed using another buffer array

You are creating a variable size array here. This will fail if n is too large (we basically run out of stack). Try to implement this without any additional allocation.

Reply:
Implemented without using a buffer array (the problem with overlapping arrays should be solved)